source/loader/layers/sanitizer/ur_sanddi.cpp

```diff
@@ -357,6 +357,7 @@ __urdlllocal ur_result_t UR_APICALL urProgramLink(
 
     UR_CALL(pfnProgramLink(hContext, count, phPrograms, pOptions, phProgram));
 
+    UR_CALL(getContext()->interceptor->insertProgram(*phProgram));
```

Comment:
This will make "insertProgram" elsewhere meaningless.
How about removing them?

Reply:
Or maybe we missed some functions need to be intercepted.

Reply:
I don't know what you mean here. Could you please explain in details?

Reply:
Now "insertProgram" in "urProgramCreateWithBinary" etc. is a little redundant.

Reply:
I think we can remove other "insertProgram"s and move "insertProgram" in to "registerProgram". Right?

Reply:
Hi @zhaomaosu , could you address @AllanZyne 's comment? Do you think this change is reasonable?

Reply:
Offline sync with maosu, this change makes sense.